### ISSUE TITLE
support for control (setpoint, power shower, sleep mode)

### ISSUE THREAD

**Issue by jonad0521** [ankohanse/hass-dab-pumps#9]
First off love the integration it was great for troubleshooting the pump and making sure pressure was staying consistent. In my quest to stop using the Dab app it would be nice if some of the inputs like power shower or setpoint could be adjusted. Are there any plans for that or will the project stay read only?

**Comment by ankohanse** [comment]
Yes good idea, and will help grow the usefulness of the integration.

Might be a couple of months before I have time to do it though.

Also I welcome any feedback on whether to expose just a select few main settings or to make available all possible inputs.

**Comment by jonad0521** [comment]
I'm not sure what possible inputs are available. To me, the most useful would be adjusting setpoints and enabling/disabling power shower and sleep mode. Maybe the duration of those as well.

**Comment by ankohanse** [comment]
In the information about all possible values available from the DAB DConnect service it includes whether a value is just for view (i.e. read-only) or can be changed. And who can change it (user, installer, support).

So for a first attempt maybe I will only allow to change the values that a user normally can change. And not the ones requiring installer rights.
When I've got an hour or so available, will lookup what list of values/inputs that will result in.

Would prefer not to use a hard-coded selection of specific values/inputs as I found that gives a hassle whenever someone starts using the integration on a pump type I hadn't seen yet.

**Comment by ankohanse** [comment]
Did a quick check and my first guess for a filter method to select 'input' candidates seems to give reasonable results:

On ESybox, parameters that can be changed by the 'customer' are:
```
- PowerShowerCommand              g=Extra Comfort
- PowerShowerBoost                g=Extra Comfort
- PowerShowerDuration             g=Extra Comfort
- SleepModeEnable                 g=Extra Comfort
- SleepModeReduction              g=Extra Comfort
- SleepModeDuration               g=Extra Comfort
- SleepModeStartTime              g=Extra Comfort
- ErasePartialFlowCounter         g=System Management  -> suppressed in custom integration
- IdentifyDevice                  g=System Management  -> suppressed in custom integration
```

On ESybox.mini, the above can be changed plus:
```
- SP_SetpointPressurePsi          g=Setpoint
- RP_PressureFallToRestartPsi     g=Setpoint
- PK_LowPressureThresholdPsi      g=Installer
- AE_AntiLock                     g=Technical Assistance
- AF_AntiFreeze                   g=Technical Assistance
- AY_AntiCycling                  g=Technical Assistance
```

Note: the 'g=' indicates the group a parameter is in. Could potentially be used to do some extra filtering.
Maybe only do 'inputs' that are in groups 'Extra Comfort' and 'Setpoint'?

**Comment by Djavdeteg** [comment]
It's not my issue, but let me add my fifty cents.
I think we need inputs, first of all, that we could add to automations, buttons, switches and so on. From this point of view, “Extra Comfort” is quite enough. Everything else is too sensitive to the system and does not change too often.

ankohanse, thank you vey much for your work. Nice integration.

**Comment by jonad0521** [comment]
Agreed
________________________________

**Comment by jonad0521** [comment]
Agreed

**Comment by ankohanse** [comment]
Thanks @djavdeteg, all feedback and ideas are welcomeOn 22 Jan 2024 16:30, jonad0521 ***@***.***> wrote:
Agreed

________________________________

From: Djavdeteg ***@***.***>

Sent: Sunday, January 21, 2024 9:29:05 PM

To: ankohanse/hass-dab-pumps ***@***.***>

Cc: jonad0521 ***@***.***>; Author ***@***.***>

Subject: Re: [ankohanse/hass-dab-pumps] support for control (setpoint, power shower, sleep mode) (Issue #9)





It's not my issue, but let me add my fifty cents.

I think we need inputs, first of all, that we could add to automations, buttons, switches and so on. From this point of view, “Extra Comfort” is quite enough. Everything else is too sensitive to the system and does not change too often.



ankohanse, thank you vey much for your work. Nice integration.

**Comment by ankohanse** [comment]
This functionality has been implemented in v2024.03.1 which has been released just now
